test/commute.js

```diff
@@ -1,9 +1,16 @@
 var assert = require('assert');
-var Maybe = require('./shared/Maybe');
 
+var S = require('sanctuary');
+
+var Maybe = require('./shared/Maybe');
 var R = require('..');
 
 
+var eq = function(actual, expected) {
+  assert.strictEqual(R.toString(actual), R.toString(expected));
+};
```

Comment:
@CrossEye, would you be happy to use this function in place of `assert.strictEqual` and `assert.deepEqual` throughout the test suite (except for testing `R.toString` itself)? If so, I submit a pull request to add __test/shared/eq.js__.

`assert.deepEqual` is horribly broken: `[]` and `{}` are considered equal, as are `[1, 2, 3]` and `['1', '2', '3']`. It's no good for comparing values of ADTs, either, since `Left(42)` and `Right(42)` may have the same internals. In such cases one is forced to use `R.equals` with `assert.strictEqual`, but then in the case of failure one sees

    AssertionError: false === true

rather than a descriptive message such as

    AssertionError: 'Left(42)' === 'Right(42)'